exercise1/exc1/package2/scripts/listener.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+import rospy
+from std_msgs.msg import UInt16, Float32
+
+last_data = None
+
+
+def callback(data):
+    global last_data
+    last_data = data.data
+
+
+def listen_and_publish(q=0.15):
+    rospy.init_node("nodeB", anonymous=True)
+
+    pub = rospy.Publisher("kthfs/result", Float32, queue_size=10)
+    rospy.Subscriber("von_hacht", UInt16, callback)
+
+    rate = rospy.Rate(20)
+    while not rospy.is_shutdown():
+        # Wait for data before pubishing.
+        if last_data is not None:
+            data = last_data / q
+            pub.publish(data)
+            rospy.loginfo("Publishing: {0}".format(data))
+            rate.sleep()
```

Comment:
```suggestion
        if last_data is None:
            continue
            
        data = last_data / q
        pub.publish(data)
        rospy.loginfo("Publishing: {0}".format(data))
        rate.sleep()
```

This is a personal preference, but I like to remove as much indentation levels as possible with this small trick :smile:

Reply:
By the way, checking if last_data is not `None` it's an interesting approach, but it will only be valid until you receive data for the first time, because after that last_data will never be `None` again.

What about clearing the variable again after publish? Something like:

```
data = last_data / q
pub.publish(data)
rospy.loginfo("Publishing: {0}".format(data))
last_data = None
rate.sleep()
```